include/hwloc/inlines.h

```diff
@@ -111,9 +111,13 @@ static __hwloc_inline const char *
 hwloc_obj_get_info_by_name(hwloc_obj_t obj, const char *name)
 {
   unsigned i;
-  for(i=0; i<obj->infos_count; i++)
+  for(i=0; i<obj->infos_count; i++) {
+    if (!obj->infos[i].name)
+      return NULL;
```

Comment:
This should never occur. We should drop entries whose name==NULL instead of inserting them.